deepchecks/checks/distribution/whole_dataset_drift.py

```diff
@@ -0,0 +1,234 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains the domain classifier drift check."""
+from functools import partial
+from typing import List
+
+import numpy as np
+import pandas as pd
+from sklearn.metrics import roc_auc_score
+from sklearn.preprocessing import OrdinalEncoder
+
+from deepchecks import Dataset, CheckResult, TrainTestBaseCheck, ConditionResult
+from deepchecks.checks.distribution.dist_utils import preprocess_for_psi
+from deepchecks.checks.distribution.plot import plot_density
+from deepchecks.utils.features import calculate_feature_importance
+from deepchecks.utils.strings import format_percent, format_number
+from deepchecks.utils.typing import Hashable
+
+from sklearn.pipeline import Pipeline
+from sklearn.impute import SimpleImputer
+from sklearn.compose import ColumnTransformer
+from sklearn.experimental import enable_hist_gradient_boosting  # noqa
+from sklearn.ensemble import HistGradientBoostingClassifier
+import matplotlib.pyplot as plt
+
+__all__ = ['WholeDatasetDrift']
+
+
+class WholeDatasetDrift (TrainTestBaseCheck):
+    """
+        Calculate drift between train and test datasets as a whole, using a model trained to distinguish between them.
+
+        Check trains a model to distinguish between train and test datasets, called a domain classifier. Then,
+        it calculates the feature importance for the domain classifier model. The result of the check is based on the
+        AUC of the domain classifier model, and the check displays the change in distribution between train and test
+        for the top features according to the calculated feature importance.
+
+        Args:
+            n_top_features (int):
+                Amount of columns to show ordered by domain classifier feature importance.
+            min_feature_importance (float): Minimum feature importance to show in the check display.
+            max_num_categories (int):
+                Only for categorical columns. Max number of allowed categories. If there are more,
+                they are binned into an "Other" category in the display. If max_num_categories=None, there is no limit.
+            sample_size (int):
+                Amount of rows to use from each dataset for the training of the domain classifier.
```

Comment:
Max num, not amount

Reply:
Also, you only use 70% of that sample to train, and the rest to test, so the wording here is inaccurate